resolver/type_syntax/type_syntax.cc

```diff
@@ -848,7 +890,15 @@ optional<TypeSyntax::ResultType> interpretTCombinator(core::Context ctx, const a
 
         default:
             if (auto e = ctx.beginError(send.loc, core::errors::Resolver::InvalidTypeDeclaration)) {
-                e.setHeader("Unsupported method `{}`", "T." + send.fun.show(ctx));
+                if (send.numPosArgs() > 0 && send.onlyPosArgs() && send.block() == nullptr && send.argsLoc().exists() &&
+                    ctx.locAt(send.funLoc).adjustLen(ctx, -1, 1).source(ctx) == ":") {
+                    auto replacement =
+                        fmt::format("T::{}[{}]", send.fun.show(ctx), ctx.locAt(send.argsLoc()).source(ctx).value());
```

Comment:
IIUC, this `argsLoc` encompasses all the args, but we don't have any multi-arg testcases (all the testcases right now are single-arg generics).  WDYT about adding some `T::Hash`-related testcases to make sure we do the right thing with multi-arg generics?

Reply:
good call thanks!

technically it conflicts with another autocorrect to remove what Sorbet sees as a redundant argument to the `Kernel#Hash` method, but the user can choose which autocorrect to pick, so it luckily just works.